packages/react/src/components/InAppMessaging/Backdrop/withBackdrop.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+import { Flex } from '../../../primitives/Flex';
+import { View } from '../../../primitives/View';
+
+import { Backdrop } from './Backdrop';
+import { BackdropProps } from './types';
+
+export function withBackdrop<Props>(
+  Content: (props: Props) => JSX.Element,
+  options: BackdropProps = {}
+): (props: Props) => JSX.Element {
+  return function ContentWithBackdrop(props: Props) {
+    return (
+      <>
+        <Backdrop {...options} />
+        <Flex className="amplify-inappmessaging-backdrop-content-container">
+          <View
+            aria-modal="true"
+            role="dialog"
```

Comment:
Are all in app messages going to use withBackdrop? If not, these should live on whatever is essentially passed as <Content />. For mobile-ey sizes, I think I saw the intent is to have a full screen view of the message. That would still match this aria pattern since it is content overlaid on top of the UI.

Reply:
Not all of them will be using it, but the ones that don't wouldn't fit the dialog/modal pattern anyway. But I can see your point about the pattern still being applicable to full screen non-backdropped versions of this as well. I'll remove this and move it into a prop for the shareable layout.